fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -3237,6 +3267,49 @@ private CompletableFuture<RecordStoreState> loadRecordStoreStateInternalAsync(@N
         return context.instrument(FDBStoreTimer.Events.LOAD_RECORD_STORE_STATE, storeHeaderFuture.thenCombine(loadIndexStates, RecordStoreState::new));
     }
 
+    @Nonnull
+    @SuppressWarnings("PMD.CloseResource")
+    public static CompletableFuture<Map<String, IndexState>> loadIndexStatesAsync(@Nonnull Subspace subspace, @Nonnull FDBRecordContext context) {
```

Comment:
It would be better to have the non-static version call this version, and remove the duplication of logic here.
Also, this method should have a javadoc and probably be marked as `INTERNAL`

Reply:
Particularly of note for the javadoc is that readable indexes are not included in the returned map